Sources/SwiftParser/Lexer/Cursor.swift

```diff
@@ -331,11 +331,35 @@ extension Lexer.Cursor {
   /// If the current character is `matching`, advance the cursor and return `true`.
   /// Otherwise, this is a no-op and returns `false`.
   mutating func advance(matching: CharacterByte) -> Bool {
-    guard self.peek() == matching.value else {
+    if self.peek(matches: matching) {
+      _ = self.advance()
+      return true
+    } else {
+      return false
+    }
+  }
+
+  /// If the current character is `matching`, advance the cursor and return `true`.
+  /// Otherwise, this is a no-op and returns `false`.
+  mutating func advance(matching character1: CharacterByte, _ character2: CharacterByte) -> Bool {
+    if self.peek(matches: character1) || self.peek(matches: character2) {
+      _ = self.advance()
+      return true
+    } else {
+      return false
+    }
+  }
+
+  /// If the current character is in `matching`, advance the cursor and return `true`.
+  /// Otherwise, this is a no-op and returns `false`.
+  @_disfavoredOverload  // favor the stamped out copies
+  mutating func advance(matching characters: CharacterByte...) -> Bool {
```

Comment:
What's the max number of characters we pass here (and similar for the peek case)? I'd prefer to just not have this one at all, I don't think we ever want to pay the cost of the ARC traffic.

Reply:
It’s three. I just stamped out three copies.